consensus/src/processes/transaction_validator/tx_validation_in_header_context.rs

```diff
@@ -0,0 +1,81 @@
+use kaspa_consensus_core::tx::Transaction;
+
+use crate::constants::LOCK_TIME_THRESHOLD;
+
+use super::{
+    errors::{TxResult, TxRuleError},
+    TransactionValidator,
+};
+
+pub(crate) enum LockTimeType {
+    Finalized,
+    DaaScore,
+    Time,
```

Comment:
can we rename this to MedianTime?

Reply:
I'm actually not sure. Median time is merely the way to get a consensus-deterministic estimation of block time (without depending on block timestamp), but the conceptual lock time type is simply "time"